src/infrastructure/app.py

```diff
@@ -21,7 +21,10 @@
 )
 
 # Declare your middlewares here
-middlewares: Iterable[Type[BaseHTTPMiddleware]] = (ErrorsHandlingMiddleware,)
+middlewares: Iterable[Type[BaseHTTPMiddleware]] = (
+    middlewares_.ErrorsHandlingMiddleware,
+    middlewares_.DatabaseTransactionMiddleware,
+)
```

Comment:
Should we use `DatabaseTransactionMiddleware` in runtime?
Cause if we use it only for tests I don't think that we should add it to the middleware.

Reply:
It is not only for test, it is used to open and close session and forcibly commit it. Inside you can use transaction(save points) to leave for last commit(Which is in middleware) or rollback then commit.

Reply:
Pay attention to files:
1. /src/middlewares/transaction.py
2. /src/infrastructure/database/core.py

Reply:
Oh, okay